background/get-userscripts.js

```diff
@@ -17,61 +17,104 @@ function getL10NURLs() {
   return urls;
 }
 
+scratchAddons.localEvents.addEventListener("addonEnabled", ({ detail }) => {
+  const { addonId, manifest } = detail;
+  chrome.tabs.query({}, (tabs) =>
+    tabs.forEach((tab) => {
+      if (tab.url || (!tab.url && typeof browser !== "undefined")) {
+        chrome.tabs.sendMessage(tab.id, "getInitialUrl", { frameId: 0 }, async (res) => {
+          if (res) {
+            const { userscripts, userstyles } = await getAddonData({ addonId, url: res, manifest });
+
+            console.log(userscripts, userstyles);
+            chrome.tabs.sendMessage(
+              tab.id,
+              {
+                dynamicAddonEnabled: {
+                  scripts: userscripts,
+                  userstyles,
+                  addonId,
+                  injectAsStyleElt: manifest.injectAsStyleElt,
+                  index: scratchAddons.manifests.findIndex((addon) => addon.addonId === addonId),
+                },
+              },
+              { frameId: 0 }
+            );
+          }
+        });
+      }
+    })
+  );
+});
+scratchAddons.localEvents.addEventListener("addonDisable", ({ detail }) => {
+  const { addonId, manifest } = detail;
+  chrome.tabs.query({}, (tabs) =>
+    tabs.forEach((tab) => {
+      if (tab.url || (!tab.url && typeof browser !== "undefined")) {
+        chrome.tabs.sendMessage(tab.id, "getInitialUrl", { frameId: 0 }, async (res) => {
```

Comment:
(Duplicate) I'm not sure Firefox likes async callbacks, I had trouble with them once